### PR TITLE
Add manageiq-providers-openstack job for public clouds

### PR DESCRIPTION
**Pull request by liu-sheng**
For #theopenlab/openlab/issues/81

### CODE REVIEW

File: playbooks/manageiq-providers-openstack-test-public-clouds/run.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
+- hosts: all
+  become: yes
+  roles:
+    - export-cloud-openrc
+  tasks:
+    - name: Run Integration tests of Docker machine against devstack
```
Comment (h00130372) [inline playbooks/manageiq-providers-openstack-test-public-clouds/run.yaml]: please modify the task name, it's tests of manageiq not Docker
machine.

File: playbooks/manageiq-providers-openstack-test-public-clouds/run.yaml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      shell:
```
Comment (h00130372) [inline playbooks/manageiq-providers-openstack-test-public-clouds/run.yaml]: can we put these cmd into a role which doing manageiq developer setup?

File: playbooks/manageiq-providers-openstack-test-public-clouds/run.yaml (hunk 1 above)
Comment (liu-sheng) [reply playbooks/manageiq-providers-openstack-test-public-clouds/run.yaml]: done, thx

File: playbooks/manageiq-providers-openstack-test-public-clouds/run.yaml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+          - liberty_keystone_v3:
+              ip: $auth_netloc
+              password: $OS_PASSWORD
+              user: $OS_USERNAME
+          EOF
+          set -x
+
+          bin/setup
+          bundle install
+          cp openstack_environments.yml spec/manageiq/
```
Comment (kiwik) [inline playbooks/manageiq-providers-openstack-test-public-clouds/run.yaml]:
```
          cp openstack_environments.yml spec/manageiq/

          pushd spec/manageiq/
          sed -i "/admin_role =/ s|:domain_id => domain_id||"  spec/tools/environment_builders/openstack/services/identity/builder/keystone_v3.rb
          ln -s $(realpath ../../spec/models/manageiq/providers/openstack) spec/models/manageiq/providers/openstack
          bundle install
          bundle exec rake db:setup
          popd
```
These lines build OpenStack resources in devstack, if we use public cloud, not need these.

File: playbooks/manageiq-providers-openstack-test-public-clouds/run.yaml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+          wget http://ftp.cz.debian.org/debian/pool/main/o/openssl1.0/libssl1.0.2_1.0.2o-1_amd64.deb
+          apt remove -y libssl-dev
+          dpkg -i libssl1.0-dev_1.0.2o-1_amd64.deb libssl1.0.2_1.0.2o-1_amd64.deb
+
+          apt-get install -y postgresql python-psycopg2 libpq-dev
+
+          set +x
+          # Prepare openstack_environments.yml
+          auth_netloc=$(python -c "import urlparse;print urlparse.urlparse('$OS_AUTH_URL').netloc")
+          cat << EOF >> openstack_environments.yml
```
Comment (kiwik) [inline playbooks/manageiq-providers-openstack-test-public-clouds/run.yaml]: we should update `user`, `domain_id` and `port` in refresher_rhos_liberty_keystone_v3_spec.rb for public cloud.

File: playbooks/manageiq-providers-openstack-test-public-clouds/run.yaml (hunk 2 above)
Comment (kiwik) [reply playbooks/manageiq-providers-openstack-test-public-clouds/run.yaml]: +1

**Pull request closed without merging**